### PR TITLE
(2.9.x) DDF-2506 Use LatLonType for geometry sort keys

### PR DESCRIPTION
**Pull request by pklinef**
#### What does this PR do?
Backport of https://github.com/codice/ddf/pull/1307

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@pklinef

### CODE REVIEW

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java [hunk 1]
```diff
@@ -260,11 +269,25 @@ public void addFields(Metacard metacard, SolrInputDocument solrInputDocument)
...
+                                    formatIndexName + SchemaFields.SORT_KEY_SUFFIX,
+                                    createCenterPoint(attributeValues));
+                        } else if (!(AttributeFormat.BINARY.equals(format)
+                                || AttributeFormat.OBJECT.equals(format))) {
+                            solrInputDocument.addField(
+                                    formatIndexName + SchemaFields.SORT_KEY_SUFFIX,
+                                    attributeValues.get(0));
+                        }
+                    } else {
+                        if (LOGGER.isTraceEnabled()) {
```
Comment (stustison) [inline catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java]: I don't think you need this conditional.

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java [hunk 2]
```diff
@@ -900,13 +889,13 @@ private String escapeSpecialCharacters(String searchPhrase) {
...
         if (StringUtils.isNotEmpty(wkt)) {
-            String geoQuery = indexName + ":\"" + operation + "(" + wkt + ")\"";
-            return getSolrQueryWithSort(geoQuery);
+            String geoQuery = geoIndexName + ":\"" + operation + "(" + wkt + ")\"";
+            Geometry pnt = getGeometry(wkt);
+            if (pnt != null) {
+                updateDistanceSort(propertyName, pnt.getCentroid());
+            }
+            return new SolrQuery(geoQuery);
         } else {
```
Comment (stustison) [inline catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java]: You don't need this else. You could reduce the complexity by moving the If check and throw up to the top.

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java (hunk 1 above)
Comment (pklinef) [inline catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java]: Changed

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java (hunk 2 above)
Comment (pklinef) [inline catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java]: Fixed

**Comment by coyotesqrl** [comment]
Successfully heroed. 👍

**Pull request merged**